cmd/create_data_stream.go

```diff
@@ -183,14 +193,14 @@ func createDataStreamCommandAction(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func createDataStreamDescriptorFromAnswers(answers newDataStreamAnswers, packageRoot string) archetype.DataStreamDescriptor {
+func createDataStreamDescriptorFromAnswers(answers newDataStreamAnswers, packageRoot string, specVersion *semver.Version) archetype.DataStreamDescriptor {
```

Comment:
Perhaps we could unit test this function, checking the manifest it generates for given answers.

Further testing could be added to `internal/packages/archetype/data_stream_test.go`, but we wouldn't be really testing if the behaviour changes for the format version used in the package.

Reply:
I've added some unit testing as suggested, also for the survey questions i've encapsulated this logic to at least check that the question is added or not based on the version.